heap/heap.py

```diff
@@ -32,18 +34,38 @@ def _bubble_up(self, index):
             self.storage[index], self.storage[(
                 index-1)//2] = self.storage[(index-1)//2], self.storage[index]
             index = (index-1)//2
+            if index == 0:
+                break
 
     def _sift_down(self, index):
 
-        if len(self.storage) >= (2*index)+2:
-            if self.storage[(2*index)+1] > self.storage[(2*index)+2]:
-                if self.storage[(2*index)+1] > self.storage[index]:
-                    self.storage[index], self.storage[(2*index) + 1 = self.storage[(2*index)+1], self.storage[index]
-                    return 2*index + 1
-            else:
-                if self.storage[2*index+2] > self.storage[index]:
-                    self.storage[index], self.storage[2*index +
-                                                      2] = self.storage[2*index+2], self.storage[index]
-                    return 2*index + 2
+        # see which children to compare
+        if len(self.storage) < (2*index)+1:  # there is no child
```

Comment:
You want to repeat the sift down operation until the child index is out of bounds, so some kind of while loop would be good here.

Reply:
yup.  fixed